screens/HomeScreen.tsx

```diff
@@ -0,0 +1,70 @@
+import { ReactNode, useEffect, useRef } from "react";
+import Modal, { ModalRefType } from "@components/common/Modal";
+import UserPostReplyScreen from "./UserPostReplyScreen";
+
+/**
+ * A type representing a reply to view when the user opens the app
+ * from a reply notification.
+ */
+export type HomeScreenViewReply = {
+  postId: string;
+  replyId: string;
+};
+
+/**
+ * Parses a `HomeScreenViewReply` from navigation route params.
+ */
+export const homeScreenViewedReplyFromRouteParams = (
+  params: Readonly<object | undefined>
+): HomeScreenViewReply | undefined => {
+  if (!params) return undefined;
+  const { postId, replyId } = params as any;
+  if (!replyId || !postId) return undefined;
+  return {
+    postId: postId,
+    replyId: replyId,
+  };
+};
+
+export type HomeScreenProps = {
+  feedView: ReactNode;
+  replyView?: (
+    viewedReply: HomeScreenViewReply,
+    onDismissed: () => void
+  ) => ReactNode;
+  viewedReply?: HomeScreenViewReply;
+};
+
+const renderReplyScreen = (
+  viewedReply: HomeScreenViewReply,
+  onDismissed: () => void
+) => <UserPostReplyScreen {...viewedReply} onDismiss={onDismissed} />;
+
+/**
+ * The main screen when the app opens.
+ */
+const HomeScreen = ({
+  feedView,
+  viewedReply,
+  replyView = renderReplyScreen,
+}: HomeScreenProps) => {
+  const modalRef = useRef<ModalRefType>(null);
+
+  useEffect(() => {
+    if (!viewedReply) return;
+    modalRef.current?.showModal();
+  }, [viewedReply]);
+
+  return (
+    <>
+      {feedView}
+      {viewedReply && replyView && (
+        <Modal ref={modalRef}>
```

Comment:
Will the activities screen use a similar structure to render replies?

Reply:
Possibly. One thing I think we should keep in mind is that we may have more of these kinds of notifications (eg. a simple one to let you know there's an invite. or someone liked your post), which would require some scenarios where we only need to display a post without a reply.

At that point we should then probably decide on how these screens take in potentially multiple props that are mutually exclusive and can show different notifications. (I think that a similar approach to that `homeScreenViewReplyFromRouteParams` function may not be such a bad idea in that case).